src/realm/geospatial.hpp

```diff
@@ -176,6 +237,99 @@ class Geospatial {
 
     mutable std::shared_ptr<S2Region> m_region;
     S2Region& get_region() const;
+
+    friend class GeospatialRef;
+};
+
+template <>
+inline GeoCenterSphere Geospatial::get<GeoCenterSphere>() const noexcept
+{
+    REALM_ASSERT_EX(m_type == Type::CenterSphere, get_type_string());
+    REALM_ASSERT(is_radius_valid());
+    REALM_ASSERT(m_points.size() >= 1);
+    return GeoCenterSphere{m_radius_radians, m_points[0]};
+}
+
+template <>
+inline GeoPolygon Geospatial::get<GeoPolygon>() const noexcept
+{
+    REALM_ASSERT_EX(m_type == Type::Polygon, get_type_string());
+    REALM_ASSERT(m_points.size() >= 1);
+    return GeoPolygon(m_points);
+}
+
+class GeospatialRef {
```

Comment:
Seems like this works by coincidence partially - Mixed assumes default trivial constructor and assignment operator, so this type is copied in many placed on Mixed initialisation and usage. Would then this be easier to just store pointer for GeospatialStorage (with all these parameters) since it's really tied to its lifetime anywat?